Examples/SnapinsSDKKotlinExample/app/src/javaImplDebug/java/com/salesforce/snapinssdkexample/SupportHomeViewAddition.java

```diff
@@ -0,0 +1,236 @@
+import com.salesforce.android.sos.api.Sos;
+import com.salesforce.snapinssdkexample.activities.settings.ChatSettingsActivity;
+import com.salesforce.snapinssdkexample.utils.ServiceSDKUtils;
+import com.salesforce.snapinssdkexample.utils.Utils;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import io.github.yavski.fabspeeddial.FabSpeedDial;
+import io.github.yavski.fabspeeddial.SimpleMenuListenerAdapter;
+
+import static com.salesforce.snapinssdkexample.utils.Utils.getBooleanPref;
+
+/**
+ * An addition to display Knowledge. Cases, Chat and SOS are launched from here as an example of
+ * how these snap-ins can be displayed within an existing application.
+ */
+
+public class SupportHomeViewAddition implements KnowledgeViewAddition{
+    private Context context;
+
+    /**
+     * Creates the view addition.
+     */
+    @NonNull
+    @Override
+    public View createView(ViewGroup viewGroup, Context context) {
+        return LayoutInflater.from(context).inflate(R.layout.content_support_home, viewGroup, false);
+    }
+
+    /**
+     * Initializes the view addition and listens for menu activity.
+     */
+    @Override
+    public void initView(View view, boolean visible) {
+        final FabSpeedDial fab = view.findViewById(R.id.support_home_fab);
+
+        fab.setMenuListener(new SimpleMenuListenerAdapter(){
+            @Override
+            public boolean onMenuItemSelected(MenuItem menuItem) {
+                switch (menuItem.getItemId()) {
+                    case R.id.action_case:
+                        launchCases();
+                        break;
+                    case R.id.action_chat:
+                        launchChat();
+                        break;
+                    case R.id.action_sos:
+                        launchSos();
+                        break;
+                }
+                return true;
+            }
+        });
+
+        if (!visible) {
+            fab.setScaleX(0);
+            fab.setScaleY(0);
+        }
+
+        context = fab.getContext();
+
+    }
+
+
+    /**
+     * Determines which scenes should have the view addition.
+     */
+    @Override
+    public boolean visibleFor(KnowledgeScene knowledgeScene) {
+        return (knowledgeScene == KnowledgeScene.SCENE_HOME);
+    }
+
+    /**
+     * Returns the entrance animator object.
+     */
+    @NonNull
+    @Override
+    public Animator getEnterAnimator(View view) {
+        return getAnimatorSet(view, 1f);
+    }
+
+    /**
+     * Returns the exit animator object.
+     */
+    @NonNull
+    @Override
+    public Animator getExitAnimator(View view) {
+        return getAnimatorSet(view, 0f);
+    }
+
+    private Animator getAnimatorSet(View view, Float value){
+        View fab = view.findViewById(R.id.support_home_fab);
+
+        AnimatorSet set = new AnimatorSet();
+        set.playTogether(
+                ObjectAnimator.ofFloat(fab, "scaleX", value),
```

Comment:
Use the constants from `View` instead? `View.SCALE_X` etc.

Reply:
done